x-pack/plugins/ml/common/types/security.ts

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const REQUIRED_ROLES = ['machine_learning_admin', 'machine_learning_user'];
```

Comment:
These three arrays aren't used, what are they for?
There are also data frame roles which might be required. If these are going to be used to grant access to something, should we instead be relying on the underlying privileges rather than the roles? As new roles which are identical to these could be created and so should grant the same access.